src/history/file_backed.rs

```diff
@@ -191,13 +191,26 @@ impl History for FileBackedHistory {
         Ok(())
     }
 
-    fn delete(&mut self, _h: super::HistoryItemId) -> Result<()> {
-        Err(ReedlineError(
-            ReedlineErrorVariants::HistoryFeatureUnsupported {
-                history: "FileBackedHistory",
-                feature: "removing entries",
-            },
-        ))
+    fn delete(&mut self, h: super::HistoryItemId) -> Result<()> {
+        let id = usize::try_from(h.0).map_err(|_| {
+            ReedlineError(ReedlineErrorVariants::OtherHistoryError(
+                "Invalid ID (not usize)",
+            ))
+        })?;
+        if self.len_on_disk <= id {
+            self.entries.remove(id);
+            Ok(())
+        } else {
```

Comment:
I think this really needs to be covered by tests. The `len_on_disk` starts to get interesting when we need to trim the buffer and other sessions are potentially having a say during `.sync()`